be/src/exec/vectorized/olap_meta_scanner.cpp

```diff
@@ -50,12 +38,31 @@ Status OlapMetaScanner::get_chunk(RuntimeState* state, ChunkPtr* chunk) {
     return _reader->do_get_next(chunk);
 }
 
-Status OlapMetaScanner::close(RuntimeState* state) {
+Status OlapMetaScanner::open(RuntimeState* state) {
+    DCHECK(!_is_closed);
+    if (!_is_open) {
+        _is_open = true;
+        _runtime_state = state;
+        RETURN_IF_ERROR(_get_tablet(_params.scan_range));
```

Comment:
we get tablet when scanner open, so if we don't open at once，there will be a `failed to get tablet` issue if compaction happen

Reply:
Now capture_consistent_rowsets will be called in MetaReader::init stage